modules/dnn/src/dnn.cpp

```diff
@@ -3886,13 +3891,8 @@ void Net::setInput(InputArray blob, const String& name, double scalefactor, cons
     bool oldShape = prevShape == blobShape;
 
     blob_.copyTo(impl->netInputLayer->inputsData[pin.oid]);
-    if (!oldShape) {
+    if (!oldShape)
```

Comment:
I think it's better to leave brackets (since `updateLayerShapes()` is enclosed by them)